src/server/enterprise/server/enterprise_test.go

```diff
@@ -26,138 +26,92 @@ func TestGetState(t *testing.T) {
 }
 
 func TestGetActivationCode(t *testing.T) {
 	if testing.Short() {
 		t.Skip("Skipping integration tests in short mode")
 	}
+	testutil.DeleteAll(t)
+	defer testutil.DeleteAll(t)
 	client := testutil.GetPachClient(t)
 
-	// Activate Pachyderm Enterprise and make sure the state is ACTIVE
-	_, err := client.Enterprise.Activate(context.Background(),
-		&enterprise.ActivateRequest{ActivationCode: testutil.GetTestEnterpriseCode(t)})
+	testutil.ActivateEnterprise(t, client)
+
+	resp, err := client.Enterprise.GetActivationCode(client.Ctx(), &enterprise.GetActivationCodeRequest{})
 	require.NoError(t, err)
-	require.NoError(t, backoff.Retry(func() error {
-		resp, err := client.Enterprise.GetActivationCode(context.Background(),
-			&enterprise.GetActivationCodeRequest{})
-		if err != nil {
-			return err
-		}
-		if resp.State != enterprise.State_ACTIVE {
-			return errors.Errorf("expected enterprise state to be ACTIVE but was %v", resp.State)
-		}
-		expires, err := types.TimestampFromProto(resp.Info.Expires)
-		if err != nil {
-			return err
-		}
-		if time.Until(expires) <= year {
-			return errors.Errorf("expected test token to expire >1yr in the future, but expires at %v (congratulations on making it to 2026!)", expires)
-		}
-		if resp.ActivationCode != testutil.GetTestEnterpriseCode(t) {
-			return errors.Errorf("incorrect activation code, got: %s, expected: %s", resp.ActivationCode, testutil.GetTestEnterpriseCode(t))
-		}
-		return nil
-	}, backoff.NewTestingBackOff()))
+	require.Equal(t, testutil.GetTestEnterpriseCode(t), resp.ActivationCode)
+	require.Equal(t, resp.State, enterprise.State_ACTIVE)
 
 	// Make current enterprise token expire
 	expires := time.Now().Add(-30 * time.Second)
 	expiresProto, err := types.TimestampProto(expires)
 	require.NoError(t, err)
-	_, err = client.Enterprise.Activate(context.Background(),
-		&enterprise.ActivateRequest{
+	_, err = client.License.Activate(client.Ctx(),
+		&lc.ActivateRequest{
 			ActivationCode: testutil.GetTestEnterpriseCode(t),
 			Expires:        expiresProto,
 		})
 	require.NoError(t, err)
-	require.NoError(t, backoff.Retry(func() error {
-		resp, err := client.Enterprise.GetActivationCode(context.Background(),
-			&enterprise.GetActivationCodeRequest{})
-		if err != nil {
-			return err
-		}
-		if resp.State != enterprise.State_EXPIRED {
-			return errors.Errorf("expected enterprise state to be EXPIRED but was %v", resp.State)
-		}
-		respExpires, err := types.TimestampFromProto(resp.Info.Expires)
-		if err != nil {
-			return err
-		}
-		if expires.Unix() != respExpires.Unix() {
-			return errors.Errorf("expected enterprise expiration to be %v, but was %v", expires, respExpires)
-		}
-		if resp.ActivationCode != testutil.GetTestEnterpriseCode(t) {
-			return errors.Errorf("incorrect activation code, got: %s, expected: %s", resp.ActivationCode, testutil.GetTestEnterpriseCode(t))
-		}
-		return nil
-	}, backoff.NewTestingBackOff()))
+	_, err = client.Enterprise.Activate(client.Ctx(),
```

Comment:
Just checking my understanding here, does the call to `Enterprise.Activate` force the enterprise server to refresh the license cache?

Reply:
Yes, exactly.